pkg/scalers/prometheus_scaler.go

```diff
@@ -223,14 +243,20 @@ func (s *prometheusScaler) ExecutePromQuery(ctx context.Context) (float64, error
 
 	// allow for zero element or single element result sets
 	if len(result.Data.Result) == 0 {
-		return 0, nil
+		if s.metadata.ignoreNullValues {
+			return 0, nil
+		}
+		return -1, fmt.Errorf("prometheus metrics %s target maybe lost, the result is empty", s.metadata.metricName)
```

Comment:
I think this is more accurate, if I'm not mistaken?
```suggestion
		return -1, fmt.Errorf("prometheus metrics %s target may be lost, the result is empty", s.metadata.metricName)
```

Reply:
update